### DISCUSSION TITLE
Primary key

### DISCUSSION THREAD

**Discussion by Akzqwerty** [simolus3/drift#1586] [Q&A] [answered]
Hi guys, i have a question. In my json i do not have an id field or a field with unique value(most of them are repeated). And i cannot compile my app without having a primary key. Is there a way to create an id for each json object inside table and assign it, or any other way i can have primary key without unique field?

**Comment by frankvollebregt** [comment, marked as answer]
If you look at the 'declaring tables' section of the get started page [link](https://drift.simonbinder.eu/docs/getting-started/#declaring-tables), you can declare an id which auto-increments yourself.
```dart
class Todos extends Table {
  IntColumn get id => integer().autoIncrement()();  // <-- this line right here
  TextColumn get title => text().withLength(min: 6, max: 32)();
  TextColumn get content => text().named('body')();
  IntColumn get category => integer().nullable()();
}
```
This way, any time an entry is pushed, an id should be incremented and set for it.
